### PR TITLE
Post Testing.

### PR DESCRIPTION
**Pull request by monocliffe**
[TPOZ-1618](https://wpengine.atlassian.net/browse/TPOZ-1618)

Implemented unit tests for the post controller and the post model. Implemented behavioural tests using Capybara/Selenium.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -65,6 +65,6 @@ def load_post
 
     flash[:alert] = 'Post Not found!' if @post.nil?
 
-    redirect_to posts_path
+    redirect_to posts_path if @post.nil?
```
Comment (JamesDominy) [inline app/controllers/posts_controller.rb:68]: May as well collapse this into a single if block

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -2,24 +2,20 @@
 
 RSpec.describe Post, type: :model do
   context 'validation' do
```
Comment (JamesDominy) [inline spec/models/post_spec.rb]: What is the purpose of these tests?

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (monocliffe) [reply app/controllers/posts_controller.rb:68]: I had it like that in the past and got complaints from rubocop and RubyMine

File: spec/models/post_spec.rb (hunk 2 above)
Comment (monocliffe) [reply spec/models/post_spec.rb]: To check if the post is still saved when some of the fields are blank.

**Review by tmwpengine — APPROVED** [review]
Looks good

File: spec/models/post_spec.rb (hunk 2 above)
Comment (JamesDominy) [reply spec/models/post_spec.rb]: Nothing in the expectation indicates that the fixture has blank fields and that this is what your are testing. You should name your test and your fixture more explicitly.

`expect(FactoryBot.build(:post_with_missing_optional_fields).save).to eq(true)`

Comment (monocliffe) [reply spec/models/post_spec.rb]: Thanks for the feedback. I'll change this and look through the other tests too.

**Review by JamesDominy — APPROVED** [review]

**Pull request merged**